internal/ingress/controller/nginx.go

```diff
@@ -784,6 +803,46 @@ func configureDynamically(pcfg *ingress.Configuration, port int) error {
 	return nil
 }
 
+// configureCertificates JSON encodes certificates and POSTs it to an internal HTTP endpoint
+// that is handled by Lua
+func configureCertificates(pcfg *ingress.Configuration, port int) error {
```

Comment:
let's call this function inside `configureDynamically` and also extract the POST logic into a helper function